addons/overheating/functions/fnc_coolWeaponWithItem.sqf

```diff
@@ -61,7 +61,7 @@ private _fnc_onSuccess = {
     };
 
     // cool the weapon
-    private _barrelMass = _weapon call FUNC(getBarrelMass);
+    ([_weapon] call FUNC(getWeaponData)) params ["", "", "", "", "", "", "", "_barrelMass"];
```

Comment:
```suggestion
    ([_weapon] call FUNC(getWeaponData)) param [7];
```
This might be worth macro'ing.
Also why does this no longer just call `getBarrelMass`, I don't see a reason to deprecate that function, it's convenient wrapper for this magic param index.

Reply:
wouldn't `([_weapon] call FUNC(getWeaponData)) param [7];` have to be this?

`private _barrelMass = ([_weapon] call FUNC(getWeaponData)) param [7];`

`_barrelMass` gets used on the next line in `_temperature = [_temperature, _barrelMass, _liquidAmount * 10] call FUNC(calculateCooling);`

Reply:
Ah yes, of course! Fixed in suggestion.

Reply:
Part of my change to params was because it creates the private variable automatically, as well as this being the way getting caged weapon data is retrieved in other places in Overheating. 

Is doing it the way you suggest faster? Or do you just want it for readability?

Reply:
Lots of empty arguments in `params` I believe is slower?

But idk, I still prefer that helper function. Either way, good to merge I think.

Reply:
Why `param` instead of `select` if you're not using the default value?

Reply:
0.001 ms for `private _barrelMass = ["1","2","3","4","5","6","7"] select 6;`
0.0012 ms for `private _barrelMass = ["1","2","3","4","5","6","7"] param [7];`
0.0015 ms for `["1","2","3","4","5","6","7"] params ["", "", "", "", "", "", "", "_barrelMass"];`

So it's definitely slower, I'll switch it to `select` for when only one value is needed

Reply:
Pabst merged just before I committed so I'll make another quick PR for this